Modules/posixmodule.c

```
@@ -5114,6 +5114,165 @@ enum posix_spawn_file_actions_identifier {
+                goto error;
+            else
+                break;
+        }
+
+        signum = PyLong_AsLong(item);
+        Py_DECREF(item);
+        if (signum == -1 && PyErr_Occurred())
+            goto error;
+        if (0 < signum && signum < NSIG) {
+            /* bpo-33329: ignore sigaddset() return value as it can fail
+             * for some reserved signals, but we want the `range(1, NSIG)`
+             * idiom to allow selecting all valid signals.
+             */
+            (void) sigaddset(mask, (int)signum);
+        }
+        else {
+            PyErr_Format(PyExc_ValueError,
+                         "signal number %ld out of range", signum);
+            goto error;
+        }
+    }
+    result = 0;
+
+error:
+    Py_XDECREF(iterator);
+    return result;
+}
+
+static int
+parse_posix_spawn_flags(PyObject *setpgroup, PyObject *resetids, PyObject *setsigmask,
+                        PyObject *setsigdef, PyObject *setscheduller,
+                        posix_spawnattr_t *attrp)
+{
+    long all_flags = 0;
+
+    errno = posix_spawnattr_init(attrp);
+    if (errno) {
+        posix_error();
+        return -1;
+    }
+
+    if (setpgroup != Py_None) {
+        int pgid = PyLong_AsLong(setpgroup);
+        if (pgid == -1 && PyErr_Occurred()) {
+            goto fail;
+        }
+        errno = posix_spawnattr_setpgroup(attrp, pgid);
+        if (errno) {
+            posix_error();
+            goto fail;
+        }
+
+        all_flags |= POSIX_SPAWN_SETPGROUP;
+    }
+
+    if (resetids != Py_None) {
+        if (resetids == Py_True) {
+            all_flags |= POSIX_SPAWN_RESETIDS;
+        }
+        else {
+            PyErr_SetString(PyExc_TypeError,
+            "The resetids parameter bust be either True or False.");
+            goto fail;
+        }
+    }
+
+   if (setsigmask != Py_None) {
+
+        sigset_t set;
+
+        if (iterable_to_sigset(setsigmask, &set)){
+            PyErr_SetString(PyExc_TypeError,
```

This hides any error raised by `iterable_to_sigset()`.